### PR TITLE
security: Encode Gmail batch message IDs

### PR DESCRIPTION
**Pull request by elie222**
> ⚪ **Browser QA skipped** · [QA report](https://qa.getinboxzero.com/20260812-120328_pr-3237_93b346c17925/report.html)
>
> <sub>Apply `rerun-qa` to rerun.</sub>

Prevents user-controlled Gmail message IDs from injecting additional multipart batch subrequests.

- encode each opaque message ID before interpolating it into a Gmail request path
- add a regression test covering CRLF and multipart-boundary injection
- preserve existing behavior for normal Gmail message IDs

<a href="https://cubic.dev/pr/elie222/inbox-zero/pull/3237?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/web/utils/gmail/batch.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    );
+
+    const requestBody = fetchMock.mock.calls[0]?.[1]?.body as string;
+
+    expect(requestBody).toContain(
+      `GET /gmail/v1/users/me/messages/${validId}\n\n`,
+    );
+    expect(requestBody.match(/^GET /gm)).toHaveLength(2);
+    expect(requestBody).not.toContain(
+      "\r\nGET /gmail/v1/users/me/messages/injected",
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/gmail/batch.test.ts:45]: Fixed in 93b346c17925a0c1669729628f29bab6f2343440. The regression now asserts the exact encodeURIComponent form of the malicious ID appears in the single intended request target, in addition to checking that no injected CRLF request line is present.
